File: main/examples/nodejs/cloud-run-local/seed.js
'use strict';

const { BigQuery } = require('@google-cloud/bigquery');
const { PassThroughClient } = require('google-auth-library');

const PROJECT = process.env.BQ_PROJECT || 'bqemu-demo';
const DATASET = process.env.BQ_DATASET || 'cr_demo';
const REST_URL = process.env.BQEMU_REST_URL || 'http://localhost:9050';

async function main() {
  // PassThroughClient skips ADC entirely — the documented pattern for
  // pointing the SDK at a local emulator.
  const bq = new BigQuery({
    projectId: PROJECT,
    apiEndpoint: REST_URL,
    authClient: new PassThroughClient(),
  });
  try {
    await bq.createDataset(DATASET, { location: 'US' });
  } catch (err) {
    if (!/already exists/i.test(String(err))) throw err;
  }
  const dataset = bq.dataset(DATASET);
  try {
    await dataset.createTable('customers', {
      schema: [
        { name: 'id', type: 'INTEGER' },
        { name: 'name', type: 'STRING' },
      ],
    });
  } catch (err) {
    if (!/already exists/i.test(String(err))) throw err;
  }
  await dataset.table('customers').insert([
    { id: 1, name: 'Alice' },
    { id: 2, name: 'Bob' },
    { id: 3, name: 'Carol' },
  ]);
  console.log(`seeded ${PROJECT}.${DATASET}.customers with 3 rows`);
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: 1.2.0/examples/docker-compose/full-stack/seed.py
"""Seed the demo dataset against the compose-hosted bqemulator."""

from __future__ import annotations

import os
import sys

from google.api_core.client_options import ClientOptions
from google.auth.credentials import AnonymousCredentials
from google.cloud import bigquery


def main() -> None:
    project = os.environ.get("BQ_PROJECT", "bqemu-demo")
    dataset = os.environ.get("BQ_DATASET", "full_stack_demo")
    rest = os.environ.get("BQEMU_REST_URL", "http://localhost:9050")

    client = bigquery.Client(
        project=project,
        credentials=AnonymousCredentials(),  # type: ignore[no-untyped-call]
        client_options=ClientOptions(api_endpoint=rest),
    )
    try:
        ds = bigquery.Dataset(f"{project}.{dataset}")
        ds.location = "US"
        client.create_dataset(ds, exists_ok=True)
        table = bigquery.Table(
            f"{project}.{dataset}.customers",
            schema=[
                bigquery.SchemaField("id", "INTEGER"),
                bigquery.SchemaField("name", "STRING"),
            ],
        )
        client.create_table(table, exists_ok=True)
        errors = client.insert_rows_json(
            f"{project}.{dataset}.customers",
            [
                {"id": 1, "name": "Alice"},
                {"id": 2, "name": "Bob"},
                {"id": 3, "name": "Carol"},
            ],
        )
        if errors:
            print(f"insert errors: {errors}", file=sys.stderr)
            sys.exit(1)
        print(f"seeded {project}.{dataset}.customers")
    finally:
        client.close()


if __name__ == "__main__":
    main()
